helix-term/src/ui/editor.rs

```diff
@@ -506,15 +521,15 @@ impl EditorView {
                     let space = if whitespace.render.space() == WhitespaceRenderValue::All
                         && !is_trailing_cursor
                     {
-                        &space
+                        space_char
                     } else {
                         " "
                     };
 
                     let nbsp = if whitespace.render.nbsp() == WhitespaceRenderValue::All
-                        && text.len_chars() < end
+                        && !is_trailing_cursor
```

Comment:
I realised that `text.len_chars() < end` was equivalent to the `is_trailing_cursor` variable, however for the `space` character it was negated, and here it wasn't. I thought this was intended, but when I was recording the demo I found that `nbsp` characters were disappearing when they were on lines previous to the cursor. I believe this was an overlook (which nobody notices because people don't go around using nbsp characters). I've inverted the condition to fix the issue.